src/System.Management.Automation/engine/parser/Compiler.cs

```diff
@@ -5537,6 +5543,46 @@ public object VisitArrayLiteral(ArrayLiteralAst arrayLiteralAst)
                                            arrayLiteralAst.Elements.Select(elem => Compile(elem).Cast(typeof(object))));
         }
 
+        public object VisitListLiteral(ListLiteralAst listLiteralAst)
+        {
+            if (listLiteralAst.Elements.Count == 0)
+            {
+                return ExpressionCache.NewEmptyObjectList;
+            }
+
+            if (listLiteralAst.Elements.Count == 1)
+            {
+                Expression values = Compile(listLiteralAst.Elements[0]);
+
+                if (values.Type == typeof(List<object>)) { return values; }
```

Comment:
I don't understand this.  Does this mean the following would be true?

```
$x = [1]
$y = [[1]]
$x.Count -eq $y.Count -and $x[0] -eq $y[0]
```

Reply:
Reading more of the implementation, I'm confused now. Is this supposed to emulate `ArrayExpressionAst` or `ArrayLiteralAst` - it seems like the former, but it's named like the later.

Reply:
This is a change I made that is different from `ArrayLiteral` -- when there is only one expression element in the `ListLiteral`, unwrap the value of the expression element like `@()` does.
This is because I think it may be better for `[(dir)]` to return a list containing all files than it to return a list containing an object array.
It's just one of my thoughts, not necessarily a right one 😄

Reply:
You shouldn't call it a literal if you can't guarantee the number of objects in the result.

Reply:
Good point. I will remove that part of logic.